exchanger/forwarder.go

```diff
@@ -23,13 +23,23 @@ func (f Forwarder) Forward(m *dns.Msg, proto string) (*dns.Msg, error) {
 // If no addresses or no matching protocol exchanger exist, a *ForwardError will
 // be returned.
 func NewForwarder(addrs []string, exs map[string]Exchanger) Forwarder {
+	// List of IP:port pairs from addrs with or without ports
+	normalized := make([]string, len(addrs))
+	for i, addr := range addrs {
+		host, port, err := net.SplitHostPort(addr)
+		if err != nil {
+			host = addr
+			port = "53"
```

Comment:
I agree, but to do that correctly I think we'd want to make a common package or something like that, because the `records/validation.go` would also want to see it. It seems a bit overkill to introduce something like that for this. Thoughts?